File: 백준/Silver/15650. N과 M （2）/N과 M （2）.swift
var input = readLine()!.split(separator: " ").map {Int($0)!}

func solution(maxNum: Int, count: Int) {
    var arr: [Int] = [Int](repeating: -1, count: count)
    var usedArr: [Bool] = [Bool](repeating: false, count: count)
    
    func dfs(depth: Int, inputNum: Int) {
        if depth == count {
            var printArr = ""
            for item in arr {
                printArr.append("\(item) ")
            }
            print(printArr)
            return
        }
        if inputNum <= maxNum {
        for item in inputNum...maxNum {
            arr[depth] = item
            dfs(depth: depth+1, inputNum: item+1)
        }
    }
    }
    dfs(depth: 0, inputNum: 1)
}

solution(maxNum: input[0], count: input[1])

File: 백준/Silver/15650. N과 M （2）/N과 M （2）.java
import java.util.Scanner;

public class Main {
    static int[] arr;

    public static void main(String[] args) {
        Scanner scanner=new Scanner(System.in);
        int n=scanner.nextInt();
        int m=scanner.nextInt();
        arr=new int[m];
        dfs(1,0,n);
    }
    public static void dfs(int start,int deep,int n){
        if(deep== arr.length){
            for(int item: arr)
                System.out.print(item+" ");
            System.out.println();
            return;
        }

        for(int i=start;i<=n;i++){
            arr[deep]=i;
            dfs(i+1,deep+1,n);
        }
    }
}